python/nav/ipdevpoll/neighbor.py

```diff
@@ -219,9 +219,8 @@ def _interface_from_name(self, name):
         if name.isdigit():
             queries.append(Q(baseport=int(name)))
 
-        netbox = Q(netbox__id=self.netbox.id)
         for query in queries:
-            ifc = self._interface_query(netbox & query)
+            ifc = self._interface_query(query)
```

Comment:
My apologies, I misread and therefore misinterpreted the diff on this line. Happens easily on a Friday afternoon ;)